### PR TITLE
Remove unused code & fix datatype

### PR DESCRIPTION
**Pull request by YULEMING**

### CODE REVIEW

File: src/dispenser/dispenser.h [hunk 1]
```diff
@@ -63,7 +63,7 @@ void dispenserCreate(dispenser_t *dispenser, motorAddress_t address, uint8_t dis
  * @param haltTime  the time in ms to halt
  * @return void
  */
-void dispenserSetHaltTime(dispenser_t *dispenser, uint32_t haltTime);
+void dispenserSetHaltTime(dispenser_t *dispenser, int32_t haltTime);
```
Comment (DavidFederl) [inline src/dispenser/dispenser.h:66]: The halt time can't be negative. Why did you change the type from `uint32_t` to `int32_t`?

File: src/dispenser/dispenser.c [hunk 2]
```diff
@@ -51,7 +51,7 @@ void dispenserEmergencyStop(dispenser_t *dispenser) {
     dispenser->state = (dispenserState_t){.function = &sleepState};
 }
 
-void dispenserSetHaltTime(dispenser_t *dispenser, uint32_t haltTime) {
+void dispenserSetHaltTime(dispenser_t *dispenser, int32_t haltTime) {
```
Comment (DavidFederl) [inline src/dispenser/dispenser.c:54]: The halt time can't be negative. Why did you change the type from `uint32_t` to `int32_t`?

**Review by DavidFederl — CHANGES_REQUESTED** [review]

File: src/dispenser/dispenser.c (hunk 2 above)
Comment (YULEMING) [reply src/dispenser/dispenser.c:54]: It could be negative during the countdown in topState.
```c
static dispenserState_t topState(dispenser_t *dispenser) {
    if (dispenser->haltTime > 0) {
        sleep_ms(TOP_TIME_SLOT);
        dispenser->haltTime = dispenser->haltTime - dispenser->dispensersInTopState;
        return topState_t;
    }
```
Also, I use the condition: dispenser->haltTime > 0 to check.
`dispenser->haltTime` won't always reduce to 0. Usually, it would be less than 0. But using `uint32_t` will underflow.

Comment (DavidFederl) [reply src/dispenser/dispenser.c:54]: The halt time should never be negative! The struct should only hold a uint32_t. The set function should also assure that.
When you change the value inside sleep state you should assure that it is only 0 and not negative, by first performing the calculation and that that this value or zero.

Comment (YULEMING) [reply src/dispenser/dispenser.c:54]: Make sense. I will change the code.

Comment (YULEMING) [reply src/dispenser/dispenser.c]: The new code is like this:
* Compare `dispenser->haltTime` and `dispenser->dispensersInTopState`
* If underflow: (`dispenser->haltTime` < `dispenser->dispensersInTopState`)
    * set haltTime to 0
* If not: do the subtraction.
Please check it.
